Extension/src/LanguageServer/configurations.ts

```diff
@@ -386,6 +395,71 @@ export class CppProperties {
         }
     }
 
+    public nodeAddonIncludesFound(): number {
+        return this.nodeAddonIncludes.length;
+    }
+
+    private async readNodeAddonIncludeLocations(rootPath: string): Promise<void> {
+        let error: Error | undefined;
+        const package_json: any = await fs.promises.readFile(path.join(rootPath, "package.json"), "utf8")
+            .then(pdj => JSON.parse(pdj))
+            .catch(e => (error = e));
+
+        const nodeAddonMap: { [dependency: string]: string } = {
+            "nan": "node --no-warnings -e \"require('nan')\"",
+            "node-addon-api": "node --no-warnings -p \"require('node-addon-api').include\""
+        };
+
+        let added: number = 0;
+        if (!error) {
+            try {
+                for (const dep in nodeAddonMap) {
+                    if (dep in package_json.dependencies) {
+                        const execCmd: string = nodeAddonMap[dep];
+                        let stdout: string = await util.execChildProcess(execCmd, rootPath);
+                        if (!stdout) {
+                            continue;
+                        }
+
+                        // cleanup newlines
+                        if (stdout[stdout.length - 1] === "\n") {
+                            stdout = stdout.slice(0, -1);
+                        }
+                        // node-addon-api returns a quoted string, e.g., '"/home/user/dir/node_modules/node-addon-api"'.
+                        if (stdout[0] === "\"" && stdout[stdout.length - 1] === "\"") {
+                            stdout = stdout.slice(1, -1);
+                        }
+
+                        // at this time both node-addon-api and nan return their own directory so this test is not really
+                        // needed. but it does future proof the code.
+                        if (!await util.checkDirectoryExists(stdout)) {
+                            // nan returns a path relative to rootPath causing the previous check to fail because this code
+                            // is executing in vscode's working directory.
+                            stdout = path.join(rootPath, stdout);
+                            if (!await util.checkDirectoryExists(stdout)) {
+                                error = new Error(`${dep} directory ${stdout} doesn't exist`);
+                                stdout = '';
+                            }
+                        }
+                        if (stdout) {
+                            added += 1;
+                            this.nodeAddonIncludes.push(stdout);
+                        }
+                    }
+                }
+            } catch (e) {
+                error = e;
+            }
+        }
+        if (error) {
+            // if it's possible for the directories returned by nan or node-addon-api to fail then
+            // this should be uncommented.
+            // console.log('readNodeAddonIncludeLocations', error.message);
```

Comment:
The logging seems like it would be okay as long as the missing package.json case is excluded (currently it's not).

Reply:
done